slybot/slybot/plugins/selectors/__init__.py

```diff
@@ -0,0 +1,28 @@
+
+class Selectors(object):
+    def setup_bot(self, settings, spec, items, extractors):
+        self.selectors = {} # { item_type: { field_name: {..} }
+
+        for template in spec['templates']:
+            selectors = self.selectors.setdefault(template['scrapes'], {})
+            selectors.update(template.get('selectors', {}))
```

Comment:
Using the selectors that match the current template is probably the safest but it needs a fallback if the annotations-plugin is disabled.

The fallback could be to try all templates and set the template used as the one that extracts the most data which is what the annotations-plugin does.